src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs

```diff
@@ -272,10 +273,11 @@ private void SetUrl()
                 }
 
                 EventSourceTrace("Url: {0}", requestUri);
+                int scopeIdIndex = 0;
                 string idnHost = requestUri.IdnHost;
-                string url = requestUri.Host == idnHost ? 
-                                requestUri.AbsoluteUri : 
-                                new UriBuilder(requestUri) { Host = idnHost }.Uri.AbsoluteUri;
+                string url = requestUri.Host == idnHost ?
```

Comment:
Another idea could be borrow some code from [here](https://source.dot.net/#System.Net.Primitives/System/Net/IPAddressParser.cs,113) or [here](https://source.dot.net/#System.Private.Uri/System/IPv6AddressHelper.cs,38) and get only host name value, if in future we fix brackets on `idnHost` this could break.